src/optimizer.cpp

```diff
@@ -231,22 +222,42 @@ auto Optimizer::getControlFromSequence(const unsigned int offset)
   return xt::view(control_sequence_.data, offset);
 }
 
-MotionModel Optimizer::getMotionModel() const
+void Optimizer::setMotionModel(const std::string & model)
 {
-  return motion_model_t_;
-}
+  if (model == "DiffDrive") {
+    motion_model_ = std::make_unique<DiffDriveMotionModel>();
+  } else if (model == "Omni") {
+    motion_model_ = std::make_unique<OmniMotionModel>();
+  } else if (model == "Ackermann") {
+    motion_model_ = std::make_unique<AckermannMotionModel>();
+  } else {
+    throw std::runtime_error(std::string(
+        "Model %s is not valid! Valid options are DiffDrive, Omni, or Ackermann", model.c_str()));
+  }
 
-void Optimizer::setMotionModel(const MotionModel motion_model)
-{
-  motion_model_t_ = motion_model;
-  state_.idx.setLayout(motion_model);
-  control_sequence_.idx.setLayout(motion_model);
+  state_.idx.setLayout(motion_model_->isHolonomic());
+  control_sequence_.idx.setLayout(motion_model_->isHolonomic());
 }
 
 xt::xtensor<double, 3> & Optimizer::getGeneratedTrajectories()
 {
   return generated_trajectories_;
 }
 
+void Optimizer::setMaxVelocities(const double vx, const double vy, const double vtheta)
+{
+  vx_max_ = vx;
+  vy_max_ = vy;
+  wz_max_ = vtheta;
+}
+
+std::array<double, 3> Optimizer::getMaxVelocities()
```

Comment:
Why do we need this ?

Reply:
The controller needs to be able to access the maximum velocities to know what the baseline starting values are to use as the basis of the speed limit callback to scale the velocities by % or reset to full speed.

Reply:
We can do this, or we can move the parameter getting of the maximum speeds to the `controller` instead of the `optimizer` and have the controller give the optimizer the max speeds to use. Then we would only need `setMaxVelocities`. I don't mind either way.